src/senselab/audio/data_structures/audio.py

```
@@ -5,14 +5,20 @@
 ease of maintaining the codebase and offering consistent public APIs.
 """
 
+try:
+    import torchaudio
+
+    TORCHAUDIO_AVAILABLE = True
+except ImportError:
```

We should catch a `ModuleNotFoundError` rather than an `ImportError`. The former is triggered only when a module isn't found, while the latter is triggered also in cases where the module IS found but can't be imported correctly for whatever one of a million reasons. Because we catch this error in order to recommend installing `torchaudio`, we do not want to catch the generic `ImportError`, as this might lead to the rare but frustrating case of `torchaudio` being installed but having an unrelated import error, which we hide and instead tell the user that it's not installed.

---

Also do you think it would be better to raise the error here rather than having a flag set which causes the error to appear later on? My inclination is to raise the error here since that's where we're catching it, but I don't have a sense of what that might entail vs raising it later on.

---

So I think raising it later helps in the case of Video where say you get an Audio object but you're not actually using it at all. I guess the idea is like you can use certain functionality without having all the dependencies, it's only when you try to use something you don't have that you raise an error. But I can see the other way working fine too

---

Nice catch. I have changed all relevant `ImportError` and made them `ModuleNotFoundError `. About when to raise the error message, for now I like it where it is because the current approach allows having some functionalities working, when they don't require the missing package

---

Sounds good. Thanks @fabiocat93 !